### PR TITLE
Node sass upgrade

### PR DESCRIPTION
**Pull request by pengyin-shan**
Fixes the previous `node-sass` not compatible issue.

It turns out that our old webpacker is locking the node-sass to version 4. So a simple upgrade of node-sass is not enough, thus few steps done to solve this issue:

1. remove `node_module` folder and `yarn.lock` file, and do `yarn cache clean` to start clean
2. update node to **12** using nvm (node-sass to supports as low as 12, node10 only compatible with node-sass <6.0)
3. update node-sass to 7 (lastest), @rails/webpacker to 5.4.3 (latest) and eslint to latest (just in case) using npm
4. run yarn install to generate new yarn.lock file

Now yarn:install passed in our test: https://github.com/portagenetwork/roadmap/runs/4437155071?check_suite_focus=true

The RSpec integration test fails but to me they are not related to this upgrade. How do you think?

*We do need to upgrade eslint our Github test to make eslint test pass*.

Please let me know if you want to adopt this approach or not. We have to upgrade node, node-sass and webpacker all together if we choose to upgrade node-sass. I think downgrade node is the other way that might not work because old webpacker lock the node-sass version in yarn.lock.

### CODE REVIEW

File: app/controllers/plan_exports_controller.rb [hunk 1]
```diff
@@ -34,7 +34,7 @@ def show
     @hash           = @plan.as_pdf(current_user, @show_coversheet)
     @formatting     = export_params[:formatting] || @plan.settings(:export).formatting
     
-    if params.key?(:phase_id)
+    if params.key?(:phase_id) && params[:phase_id].length > 0
```
Comment (lagoan) [inline app/controllers/plan_exports_controller.rb:37]: Is this left over from previous work from the multiphase download?

**Review by lagoan — CHANGES_REQUESTED** [review]
All looks good. Just a question about work from another branch.

The only thing I would change is fixing the eslint tests. Right now it is complaining about a minor difference in node version.

File: app/controllers/plan_exports_controller.rb (hunk 1 above)
Comment (pengyin-shan) [reply app/controllers/plan_exports_controller.rb:37]: yes. This change is necessary to make muti-phase download work. I found it in testing muti-phase download after node-sass upgrade

**Comment by pengyin-shan** [comment]
> All looks good. Just a question about work from another branch.
> 
> The only thing I would change is fixing the eslint tests. Right now it is complaining about a minor difference in node version.

Yeah eslint needs to be upgrade to match node 12

**Comment by lagoan** [comment]
From the workflow file, I see the action being used comes from here https://github.com/marketplace/actions/run-eslint

And we should no longer need it as it can be run directly using github actions.

From that page:

Github Actions have everything that you need to run your eslint out of the box now, to do that simply use the run commands like so:

```
name: CI
on: push
jobs:
  build:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v2
    - name: Install modules
      run: yarn
    - name: Run ESLint
      run: eslint . --ext .js,.jsx,.ts,.tsx
```

Could you make the change in the workflow along with these changes ?

**Comment by pengyin-shan** [comment]
> From the workflow file, I see the action being used comes from here https://github.com/marketplace/actions/run-eslint
> 
> And we should no longer need it as it can be run directly using github actions.
> 
> From that page:
> 
> Github Actions have everything that you need to run your eslint out of the box now, to do that simply use the run commands like so:
> 
> ```
> name: CI
> on: push
> jobs:
>   build:
>     runs-on: ubuntu-latest
>     steps:
>     - uses: actions/checkout@v2
>     - name: Install modules
>       run: yarn
>     - name: Run ESLint
>       run: eslint . --ext .js,.jsx,.ts,.tsx
> ```
> 
> Could you make the change in the workflow along with these changes ?

Wow this looks exciting. Let me try it definitely!

**Review by lagoan — APPROVED** [review]
LGTM!

**Comment by pengyin-shan** [comment]
Finally it works lol... @lagoan do you want me to upload a new image to uat?

**Comment by lagoan** [comment]
Yes please.

We should configure docker to pull new images to simplify this process.

I think there was some work done with this on UAT

Omar

+

**Pull request merged**